### PR TITLE
Issue 5124: publish :segmentstore:storage:testJar artifact

### LINKED ISSUE #5124: publish segment storage test jars
Problem description
Pravega publishes some test jars such as test-common. The segment storage test jars when published will help with testing long term storage bindings for source in other repositories. @sachin-j-joshi  prefers this and it blocks the binding project.

Problem location
build.gradle to include test jars in maven plugin.

Suggestions for an improvement
The maven plugin is a bit old but it works and all we need for this issue is to publish a test jar with test side interfaces.

### PR DESCRIPTION
**Pull request by ravibeta**
**Change log description**  
This generates a test jar so that it can be published along with the other artifacts. 

**Purpose of the change**  
Fixes #5124 

**What the code does**  
This leverages the maven plugin syntax to specify test jars for publishing. We do it the same way as [system test](https://mvnrepository.com/artifact/io.pravega/pravega-test-system) jar:
https://repo1.maven.org/maven2/io/pravega/pravega-test-system/0.7.1/

**How to verify it**  
These steps can be verified by running the gradle command to publish:
./gradlew  publish
publishing  artifact io.pravega:pravega-segmentstore-storage-tests:jar:0.9.0-2635.d0d098a-2631.017f4b1-20200831.001154-1 
to ... /io/pravega/pravega-segmentstore-storage-tests/0.9.0-2635.d0d098a-2631.017f4b1-SNAPSHOT/pravega-segmentstore-storage-tests
-0.9.0-2635.d0d098a-2631.017f4b1-20200831.001154-1.jar
Reference: https://docs.gradle.org/current/userguide/maven_plugin.html
```
io/pravega/pravega-segmentstore-storage/0.9.0-2635.d0d098a-SNAPSHOT/pravega-segmentstore-storage-0.9.0-2635.d0d098a-SNAPSHOT-javadoc.jar
io/pravega/pravega-segmentstore-storage/0.9.0-2635.d0d098a-SNAPSHOT/pravega-segmentstore-storage-0.9.0-2635.d0d098a-SNAPSHOT-sources.jar
io/pravega/pravega-segmentstore-storage/0.9.0-2635.d0d098a-SNAPSHOT/pravega-segmentstore-storage-0.9.0-2635.d0d098a-SNAPSHOT-tests.jar
 # ls -alt io/pravega/pravega-segmentstore-storage/0.9.0-2635.d0d098a-SNAPSHOT/pravega-segmentstore-storage-0.9.0-2635.d0d098a-SNAPSHOT-tests.jar
-rw-r--r-- 1 root root 212997 Aug 30 15:48 io/pravega/pravega-segmentstore-storage/0.9.0-2635.d0d098a-SNAPSHOT/pravega-segmentstore-storage-0.9.0-2635.d0d098a-SNAPSHOT-tests.jar
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1008,6 +1008,7 @@ task publishAllJars() {
     dependsOn ':shared:security'
     dependsOn ':shared:protocol:publish'
     dependsOn ':segmentstore:contracts:publish'
+    dependsOn ':segmentstore:storage:testJar'
```
Comment (ravibeta) [inline build.gradle:1011]: This is not required but it is intentional in this listing.

**Review by sachin-j-joshi — COMMENTED** [review]
The name of the test jar is 
`pravega-segmentstore-storage-0.9.0-2641.ea37ccadc-issue-5124-use-maven-plugin-tests.jar`
which I find odd 

- prefer `pravega-segmentstore-storage-tests-...jar`
what do you thing ?

**Comment by ravibeta** [comment]
Yes Sachin, that suffix is the maven convention as per the mention [here](https://docs.gradle.org/current/userguide/maven_plugin.html). 
So clients that want to declare a dependency on the test jar would use it as follows:
```
compile group:"io.pravega", name:"pravega-segmentstore-storage", version: pravegaVersion,  classifier: 'tests'
``` 
I have tried this and it works.
Please be sure to check out the before-and-after change between the links below to see that we are only introducing a test jar in the publication and keeping all other publications as earlier with no changes.
Before: https://oss.jfrog.org/artifactory/oss-snapshot-local/io/pravega/pravega-segmentstore-storage/0.9.0-2636.c431acd-SNAPSHOT/
After: https://<pravega-private-asd-server>/artifactory/<pravega-private-asd-share>-scratch-local/io/pravega/pravega-test-testcommon/0.9.0-2634.2db1b79ff-issue-5124-use-maven-plugin/

File: gradle/maven.gradle [hunk 2]
```diff
@@ -9,6 +9,15 @@
  *
  */
 plugins.withId('maven') {
+    task sourcesJar(type: Jar, dependsOn: classes) {
+         classifier = 'sources'
+         from sourceSets.main.allSource
+    }
+    artifacts{
```
Comment (andreipaduroiu) [inline gradle/maven.gradle:16]: Isn't this going to publish ALL test JARs, instead of just the one we want?

Comment (ravibeta) [reply gradle/maven.gradle:16]: Yes Andrei that is intentional and it is not visible unless users click on 'view all' as on this page: https://repo1.maven.org/maven2/io/pravega/pravega-test-testcommon/0.7.1

**Review by andreipaduroiu — APPROVED** [review]

**Pull request merged**